src/rgw/rgw_tag.cc

```diff
@@ -10,12 +10,12 @@
 #include "rgw_tag.h"
 #include "rgw_common.h"
 
-bool RGWObjTags::add_tag(const string&key, const string& val){
-  return tag_map.emplace(std::make_pair(key,val)).second;
+void RGWObjTags::add_tag(const string&key, const string& val){
+  tag_map.emplace(std::make_pair(key,val));
 }
 
-bool RGWObjTags::emplace_tag(std::string&& key, std::string&& val){
-  return tag_map.emplace(std::move(key), std::move(val)).second;
+void RGWObjTags::emplace_tag(std::string&& key, std::string&& val){
+  tag_map.emplace(std::move(key), std::move(val));
```

Comment:
am I correct that you don't need the std::move here when the argument signature is && (rvalue reference)?

Reply:
@mattbenjamin : it is explained here why we need the std::move even on a rvalue reference - https://stackoverflow.com/questions/54951635/why-to-use-stdmove-despite-the-parameter-is-an-r-value-reference.